core/services/ardupilot_manager/firmware_download/FirmwareDownload.py

```diff
@@ -20,18 +23,54 @@
     ssl._create_default_https_context = ssl._create_unverified_context
 
 
-class Vehicle(IntEnum):
+def get_current_arch() -> str:
+    """Maps paltform.machine() outputs to docker architectures"""
+    arch_map = {"x86_64": "amd64", "armv7l": "arm"}  # TODO: differentiate arm64v8/ arm32v7
```

Comment:
from https://github.com/bluerobotics/companion-docker/pull/289#discussion_r640801726:

They are different things, what are you using as reference is a docker PR that sadly we need to have track of the architecture to download the correct one based on docker API.

>  if it detects an ARM64 it will classify it as ARM and try to download the armv6/7 binary, which will not necessarily run on it.

That's not true, aarch64 can run both armv6 and armv7.

ArduPilot only deploys armv6 binaries, so no armv7 or aarch64, different from docker.
Also, armv6 runs in both armv7 and aarch64, so we don't need to worry.
In docker we do not support armv6 and we probably not will since some binaries are only provided in armv7.

Reply:
I usually prefer the safe route, not assuming something will run if we don't know it will, but if you are confident it's not a problem, I'm ok.